[Term Entry] Go Math Functions Dim() (#3319)
Codecademy/docs: content/go/concepts/math-functions/terms/dim/dim.md

---
Title: 'Dim()'

> Description: 'Calculate the Maximum of the Difference between Two Arguments'

Reviewer: ```suggestion
Description: 'Calculates the maximum of the difference between two arguments.'
```

Subjects:
  - 'Code Foundation'
  - 'Computer Science'
Tags:
  - 'Math'
  - 'Functions'
CatalogContent:
  - 'learn-go'
  - 'paths/computer-science'
---

The **`Dim()`** function takes two floating-point numbers (x and y), calculates their difference (difference := x - y), and then finds the maximum between this difference and 0 using the `math.Max()` function. The result is a single floating-point number that represents the maximum of the difference between x and y or 0.

## Syntax

```pseudo
difference := math.Dim(x, y)
```
The `Dim()` function returns a single value of type `float64`. This value represents the difference between the two arguments (`x`-`y`) and 0. If the difference is greater than 0, it is returned; else, 0 is returned.

Note these special cases:
- Passing `NaN` in this function will return `NaN`.
- Passing `Inf` in this function will return `NaN`.
- Passing `-Inf` in this function will return `NaN`.

## Example

> In this Go code example, we have a simple program that calculates and prints the positive difference between two numbers, 7.0 and 4.0, using the `math.Dim` function from the `math` package.

Reviewer: ```suggestion

In this Go code example, we have a simple program that calculates and prints the positive difference between two numbers, 7.0 and 4.0, using the `Dim()` function from the `math` package.
```

```go
package main
import (
"fmt"
"math"
)

func  main() {
// Define two numbers
a := 7.0
b := 4.0

// Calculate the positive difference between a and b using math.Dim()
difference := math.Dim(a, b)
fmt.Printf("The positive difference between %.1f and %.1f is %.1f\n", a, b, difference)

}
```

The above code results in the following output:

```shell
The  positive  difference  between 7 and 4 = 3
```

## Codebyte Example

> The example below is runnable and demonstrates `dim()` function.

Reviewer: ```suggestion

The example below is runnable and demonstrates the `Dim()` function.
```

---
Title: 'Dim()'
Description: 'Calculates the maximum of the difference between two arguments.'
Subjects:
  - 'Code Foundation'
  - 'Computer Science'
Tags:
  - 'Math'
  - 'Functions'
CatalogContent:
  - 'learn-go'
  - 'paths/computer-science'
---

The **`Dim()`** function takes two floating-point numbers (x and y), calculates their difference (difference := x - y), and then finds the maximum between this difference and 0 using the `math.Max()` function. The result is a single floating-point number that represents the maximum of the difference between x and y or 0.

## Syntax

```pseudo
difference := math.Dim(x, y)

> ```

Reviewer 2: add an empty line after the syntax block

> The `Dim()` function returns a single value of type `float64`. This value represents the difference between the two arguments (`x`-`y`) and 0. If the difference is greater than 0, it is returned; else, 0 is returned.

Reviewer 2: ```suggestion
The `Dim()` function returns a single value of type `float64`. If the difference (*x-y*) is greater than `0`, it is returned; else, `0` is returned.
```

Note these special cases:
- Passing `NaN` in this function will return `NaN`.

> - Passing `Inf` in this function will return `NaN`.

Reviewer 2: ```suggestion
- Passing positive or negative `Inf` in this function will return `NaN`.
```

- Passing `-Inf` in this function will return `NaN`.

## Example

In this Go code example, we have a simple program that calculates and prints the positive difference between two numbers, 7.0 and 4.0, using the `Dim()` function from the `math` package.

```go
package main
import (
"fmt"
"math"
)

func  main() {
// Define two numbers
a := 7.0
b := 4.0

// Calculate the positive difference between a and b using math.Dim()
difference := math.Dim(a, b)
fmt.Printf("The positive difference between %.1f and %.1f is %.1f\n", a, b, difference)

}
```

The above code results in the following output:

```shell
The  positive  difference  between 7 and 4 = 3
```

## Codebyte Example

The example below is runnable and demonstrates the `Dim()` function.

> ```codebyte/golang

Reviewer 2: It might be more valuable to demonstrate an example that results in 0 or NaN. This codebyte example is almost equivalent to the example above

Author: That's true

Author: Hi @caupolicandiaz is it ok if i give both examples in one codebyte ?

> ---
> Title: 'Dim()'
> 
> Description: 'Calculate the Maximum of the Difference between Two Arguments'
> 
> Subjects:
> 
> - 'Code Foundation'
> 
> - 'Computer Science'
> 
> Tags:
> 
> - 'Math'
> 
> - 'Functions'
> 
> CatalogContent:
> 
> - 'learn-go'
> 
> - 'paths/computer-science'
> ---

Reviewer: ```suggestion
---
Title: 'Dim()'
Description: 'Calculates the maximum of the difference between two arguments.'
Subjects:
  - 'Code Foundation'
  - 'Computer Science'
Tags:
  - 'Math'
  - 'Functions'
CatalogContent:
  - 'learn-go'
  - 'paths/computer-science'
---
```
Remove blank lines in metadata.

> The ****dim()**** function takes these two floating-point numbers (x and y), calculates their difference (difference := x - y), and then finds the maximum between this difference and 0 using the math.Max() function. The result is a single floating-point number that represents the maximum of the difference between x and y or 0.

Reviewer: ```suggestion

The **`Dim()`** function takes two floating-point numbers (x and y), calculates their difference (difference := x - y), and then finds the maximum between this difference and 0 using the `math.Max()` function. The result is a single floating-point number that represents the maximum of the difference between x and y or 0.
```

> **## Syntax**

Reviewer: ```suggestion
## Syntax
```

```pseudo

> func Dim(x, y float64) float64

Reviewer: ```suggestion
difference := math.Dim(x, y)
```
We like to see the function in use, not a function signature.

```

> The `Dim()` function returns a single value of type `float64`. This value represents the difference between the two arguments (x-y) and 0. If the difference is greater than 0, it is returned; else, 0 is returned.

Reviewer: ```suggestion

The `Dim()` function returns a single value of type `float64`. This value represents the difference between the two arguments (`x`-`y`) and 0. If the difference is greater than 0, it is returned; else, 0 is returned.
```

Note these special cases:

> - If you pass `NaN` in this function `Dim(a, NaN)` or `Dim(NaN, b)`, then this function will return NaN.
> - if you pass `inf` in the function `Dim(+inf,inf)` the function will return NaN.
> - If you pass `-Inf` in this function `Dim(-Inf, -Inf)`, then this function will return NaN.

Reviewer: ```suggestion
- Passing `NaN` in this function will return `NaN`.
- Passing `Inf` in this function will return `NaN`.
- Passing `-Inf` in this function will return `NaN`.
```
We avoid using first- and second-person pronouns in Docs.

> 
>   
> 
> **## Example**

Reviewer: ```suggestion

## Example

```

> ```go

Reviewer: We'd like an introductory sentence before the example summarizing what the example is doing.

Author: @SSwiniarski Hi Have made the necessary changes .

package main
import (
"fmt"
"math"
)

func  main() {
// Define two numbers
a := 7.0
b := 4.0

// Calculate the positive difference between a and b using math.Dim()
difference := math.Dim(a, b)
fmt.Printf("The positive difference between %.1f and %.1f is %.1f\n", a, b, difference)

}

> ```
> The above code results in the following output:
> 
>   
> 

Reviewer: ````suggestion
```

The above code results in the following output:

````

```shell
The  positive  difference  between 7 and 4 = 3
```

> **## Codebyte Example**
> The example below is runnable and demonstrates `dim()` function.

Reviewer: ```suggestion
## Codebyte Example

The example below is runnable and demonstrates `dim()` function.

```
